crates/reth/rpc/src/rpc.rs

```diff
@@ -42,4 +43,38 @@ where
 
         res.map_err(to_jsonrpsee_error("Failed fetching witness"))
     }
+
+    fn get_block_state_diff(&self, block_hash: B256) -> RpcResult<Option<BlockStateDiff>> {
+        self.db
+            .get_state_diff_by_hash(block_hash)
+            .map_err(to_jsonrpsee_error("Failed fetching block state diff"))
+    }
+
+    fn get_state_root_by_diffs(&self, block_number: u64) -> RpcResult<Option<B256>> {
```

Comment:
I would call this `_via_diffs`.  Saying "by" here implies that the diffs are a key we're looking up the state root by.

Reply:
done